pkg/virt-controller/services/template.go

```diff
@@ -1204,16 +1204,30 @@ func getMemoryOverhead(vmi *v1.VirtualMachineInstance) *resource.Quantity {
 
 	// Add fixed overhead for shared libraries and such
 	// TODO account for the overhead of kubevirt components running in the pod
-	overhead.Add(resource.MustParse("128M"))
+	overhead.Add(resource.MustParse("138Mi"))
 
 	// Add CPU table overhead (8 MiB per vCPU and 8 MiB per IO thread)
 	// overhead per vcpu in MiB
 	coresMemory := resource.MustParse("8Mi")
+	var vcpus int64
 	if domain.CPU != nil {
-		vcpus := hardware.GetNumberOfVCPUs(domain.CPU)
-		value := coresMemory.Value() * vcpus
-		coresMemory = *resource.NewQuantity(value, coresMemory.Format)
+		vcpus = hardware.GetNumberOfVCPUs(domain.CPU)
+	} else {
+		// In case guest CPU topology isn't set, for some reason, take value from resources request or limits.
+		resources := vmi.Spec.Domain.Resources
+		if cpuLimit, ok := resources.Limits[k8sv1.ResourceCPU]; ok {
+			vcpus = cpuLimit.Value()
+		} else if cpuRequests, ok := resources.Requests[k8sv1.ResourceCPU]; ok {
+			vcpus = cpuRequests.Value()
+		}
+	}
+
+	// if neither CPU topology nor request or limits provided, set vcpus to 1
+	if vcpus < 1 {
```

Comment:
Looks like this special case can now only happen in the `else` branch above?

Reply:
actually... GetNumberOfVCPUs can return 0 too